core/src/main/java/org/apache/druid/java/util/common/parsers/JSONFlattenerMaker.java

```diff
@@ -57,15 +57,21 @@ public class JSONFlattenerMaker implements ObjectFlatteners.FlattenerMaker<JsonN
   private final CharsetEncoder enc = StandardCharsets.UTF_8.newEncoder();
   private final boolean keepNullValues;
 
+  private final boolean discoverNestedFields;
 
-  public JSONFlattenerMaker(boolean keepNullValues)
+
+  public JSONFlattenerMaker(boolean keepNullValues, boolean discoverNestedFields)
   {
     this.keepNullValues = keepNullValues;
+    this.discoverNestedFields = discoverNestedFields;
   }
 
   @Override
   public Iterable<String> discoverRootFields(final JsonNode obj)
   {
+    if (discoverNestedFields) {
```

Comment:
can you drop a one line comment here? I am assuming its like this since each top-level field is a field of its own if we are allowing nested columns.

Reply:
do you want similar comments for all the other FlattenerMaker implementations?

Reply:
went ahead and added comments to all